gui-tk/csvImport.tcl

```diff
@@ -214,15 +214,18 @@ proc CSVWebImportDialog {} {
             csvDialog.requestRedraw
         }
     }
-       
-    set urlfile [inputUrl]
     
-    #set filename [tk_getOpenFile -filetypes {{CSV {.csv}} {All {.*}}} -initialfile [getFile $urlfile] ]  
+    # Example CSV file that Minsky can import, but to be replaced by user input
+    set initUrl "http://samplecsvs.s3.amazonaws.com/TechCrunchcontinentalUSA.csv"   
+    set urlFile [inputUrl $initUrl] 
+    puts $urlFile
     
-    set filename [getFile $urlfile]
+    # Create temporary file for downloaded CSV file
+    set filename [file tempfile [getFile $urlFile] ./*.csv]
```

Comment:
file tempfile returns an open channel, which can be used in puts and gets statements. The file is deleted when the channel is closed.

If you need the name of the file (which you probably do), then you can request it be placed in a TCL variable via an option to file tempfile.

Reply:
"man file" has all the goss on using the this command. Select the (n) category, as there's an unrelated file system command as well.

Reply:
Well, at least I know I can get the contents of a CSV file loaded into Minsky, but not via the filename itself (the latest commit dumps the contents of the CSV file into the error log). In the man page, were you referring to the following?

[https://www.manpagez.com/man/n/Tcl/](
https://www.manpagez.com/man/n/Tcl/)

I could not call it from a terminal on my BSD system, but found the above page online.

Reply:
Actually, this one.

http://www.manpagez.com/man/n/file/

Reply:
You might also find this resource rather helpful: http://tcl.tk/

Reply:
I wonder why your Ubuntu doesn't have the TCL man pages? Doesn't matter - the web resources work just as well.